src/librustc_driver/driver.rs

```diff
@@ -339,6 +344,17 @@ pub fn compile_input(
                     }
                 }
 
+                // Output Yorick debug sections into binary targets.
+                if sess.crate_types.borrow().contains(&config::CrateType::Executable) &&
+                    with_yk_debug_sections() {
```

Comment:
This block of code was being run even if we didn't request debug sections. I've added the `&& with_yk_debug_sections()` to the condition.